pkg/spanconfig/spanconfigsqltranslator/sqltranslator.go

```diff
@@ -366,70 +366,27 @@ func (s *SQLTranslator) generateSpanConfigurationsForTable(
 
 	records := make([]spanconfig.Record, 0)
 	if table.GetID() == keys.DescriptorTableID {
-		// We have some special handling for `system.descriptor` on account of
-		// it being the first non-empty table in every tenant's keyspace.
+		startKey := keys.SystemConfigSplitKey
 		if !s.codec.ForSystemTenant() {
 			// We start the span at the tenant prefix. This effectively installs
 			// the tenant's split boundary at /Tenant/<id> instead of
 			// /Tenant/<id>/Table/3. This doesn't really make a difference given
 			// there's no data within [/Tenant/<id>/ - /Tenant/<id>/Table/3),
 			// but looking at range boundaries, it's slightly less confusing
 			// this way.
-			record, err := spanconfig.MakeRecord(
-				spanconfig.MakeTargetFromSpan(roachpb.Span{
-					Key:    s.codec.TenantPrefix(),
-					EndKey: tableEndKey,
-				}), tableSpanConfig)
-			if err != nil {
-				return nil, err
-			}
-			records = append(records, record)
-		} else {
-			// The same as above, except we have named ranges preceding
```

Comment:
Perhaps reword some of this commentary to explain why for the system tenant we're choosing to split at SystemConfigSplitKey. I'll actually suggest using TableDataMin here instead (removing another occurrence of "SystemConfig" from SystemConfigSplitKey).